jina/peapods/remote.py

```diff
@@ -155,9 +158,181 @@ def set_ready(self, resp):
         super().set_ready()
 
 
-class RemoteMutablePod(RemotePea):
-    """A RemoteMutablePod that spawns a remote :class:`MutablePod`.
+def namespace_to_dict(args: Dict) -> Dict:
+    """ helper function to convert argparse.Namespace to json to be uploaded via REST """
+    pea_args = {}
+    for k, v in args.items():
+        if v is None:
+            pea_args[k] = None
+        if isinstance(v, Namespace):
+            pea_args[k] = vars(v)
+        if isinstance(v, list):
+            pea_args[k] = []
+            pea_args[k].extend([vars(_) for _ in v])
+    return pea_args
+
+
+def fetch_files_from_yaml(pea_args: Dict, logger) -> Tuple[set, set]:
+    """ helper function to fetch yaml & pymodules to be uploaded to remote """
+    def _file_adder(_file, _file_list):
+        if _file and _file.endswith(('yml', 'yaml', 'py')):
+            if Path(_file).is_file():
+                _file_list.add(_file)
+                logger.debug(f'adding file {_file} to be uploaded to remote context')
+            else:
+                logger.debug(f'file {_file} doesn\'t exist in the disk')
+
+    if 'peas' in pea_args:
+        uses_files = set()
+        pymodules_files = set()
+
+        for current_pea in pea_args['peas']:
+            for _arg in ['uses', 'uses_before', 'uses_after']:
+                _file_adder(_file=current_pea[_arg],
+                            _file_list=uses_files)
+
+            _file_adder(_file=current_pea['py_modules'],
+                        _file_list=pymodules_files)
+
+        if uses_files:
+            for current_file in uses_files:
+                with open(current_file) as f:
+                    result = ruamel.yaml.round_trip_load(f)
+
+                if 'metas' in result and 'py_modules' in result['metas']:
+                    _file_adder(_file=result['metas']['py_modules'],
+                                _file_list=pymodules_files)
+
+        return uses_files, pymodules_files
+
+
+class PodAPI:
```

Comment:
it seems to me that it deserves a new file.

Reply:
done